packages/sidecar/src/mcp/index.ts

```diff
@@ -31,26 +34,35 @@ export function createMcpInstance(buffer: MessageBuffer<Payload>) {
         };
       }
 
-      const errors = [];
-
-      for (const [_contentType, data] of envelopes) {
-        const text = data.toString("utf-8");
-        const [_, { type }, payload] = JSON.parse(`[${text.replaceAll("}\n{", "},{")}]`) as any[];
+      const errors = envelopes
+        .map(([contentType, data]) => processEnvelope({ contentType, data }))
+        .sort((a, b) => {
```

Comment:
We can probably move that `event.type === "event"` check here and save some memory and CPU time. It wold also allow us to do

```ts
const content = errors.map(payload => ({...}));
```

which should also be more efficient than `someArray.push()`.